dataduct/database/tests/test_database.py

```diff
@@ -0,0 +1,270 @@
+"""Tests for Database
+"""
+import os
+
+from unittest import TestCase
+from testfixtures import TempDirectory
+from nose.tools import assert_not_equal
+from nose.tools import eq_
+
+from ..database import Database
+from ..table import Table
+from ..view import View
+from ..sql import SqlScript
+
+
+class TestDatabase(TestCase):
+    """Tests for Database
+    """
+
+    @staticmethod
+    def _create_table(sql):
+        """Creates a table object from a SQL string
+        """
+        return Table(SqlScript(sql))
+
+    @staticmethod
+    def _create_view(sql):
+        """Creates a view object from a SQL string
+        """
+        return View(SqlScript(sql))
+
+    def test_create(self):
+        """Tests database initialization
+        """
+        table = self._create_table('CREATE TABLE test_begin (id INTEGER);')
+        database = Database(relations=[table])
+
+        # Verify that the database is constructed properly
+        eq_(database.num_tables, 1)
+        eq_(database.num_views, 0)
+        assert_not_equal(database.relation('test_begin'), None)
+
+    @staticmethod
+    def test_create_from_file():
+        """Tests database initialization from file
+        """
+        with TempDirectory() as d:
+            # Create files in the temp directory
+            d.write('test_table.sql',
+                    'CREATE TABLE test_table (session_id INTEGER);')
+            d.write('test_view.sql',
+                    'CREATE VIEW test_view AS (SELECT * FROM test_table);')
+            database = Database(files=[os.path.join(d.path, 'test_table.sql'),
+                                       os.path.join(d.path, 'test_view.sql')])
+
+            # Verify that the database is constructed properly
+            eq_(database.num_tables, 1)
+            eq_(database.num_views, 1)
+            assert_not_equal(database.relation('test_table'), None)
+            assert_not_equal(database.relation('test_view'), None)
+
+    @staticmethod
+    def test_create_from_file_no_relation():
+        """Database initialization with a file that does not create a
+        relation
+        """
+        with TempDirectory() as d:
+            # Create a file in the temp directory
+            d.write('test.sql',
+                    'SELECT * FROM test_table;')
+            try:
+                Database(files=[os.path.join(d.path, 'test.sql')])
+                assert False
+            except ValueError:
+                pass
+
+    @staticmethod
+    def test_create_two_arguments():
+        """Must create database with less than two arguments
+        """
+        try:
+            Database(relations=['test_rel'], files=['test_file'])
+            assert False
+        except ValueError:
+            pass
+
+    def test_create_duplicate_relations(self):
+        """Database initalization with duplicate relations
+        """
+        table = self._create_table(
+            'CREATE TABLE test_begin (session_id INTEGER);')
```

Comment:
make the strings that are used multiple times be initialized in a setup function.